spec/queue.js

```diff
@@ -0,0 +1,124 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import Queue from '../src/queue'
+
+chai.use(chaiChange)
+
+describe('Queue', () => {
+  'use strict'
+
+  it('exists', () => {
+    expect(Queue).to.be.a('function')
+  })
+
+  describe('enqueue()', () => {
+    it('pushes an element to the queue.', () => {
+      const queue = new Queue()
+
+      expect(() => queue.enqueue('foo'))
+        .to.alter(() => queue.length(), { from: 0, to: 1 })
+    })
+
+    it('pushes an element to the back of the queue.', () => {
+      const queue = new Queue()
+      queue.enqueue('foo')
+
+      expect(queue.back())
+        .to.equal('foo')
+    })
+  })
+
+  describe('dequeue()', ()=> {
+    it('returns and removes the front element in the queue.', () => {
+      const queue = new Queue()
+      queue.enqueue('foo')
+
+      expect(() => queue.dequeue())
+        .to.alter(() => queue.length(), {from: 1, to: 0 })
+    })
+
+    it('returns null if the queue is empty.', () => {
+      const queue = new Queue()
+
+      expect(queue.dequeue())
+        .to.be.null
+    })
+    it('removes returned element from queue', () => {
```

Comment:
Is this test description accurate? Looks like you're testing to see if `dequeue()` returns null if the queue is empty.